### PR TITLE
adding endpoint control for listings' hrefs, this s3 website redirect use.

### PR DESCRIPTION
**Pull request by pracplayopen**

### CODE REVIEW

**Comment by rufuspollock** [comment]
@pracplayopen could you summarize what this change does - the commit messages e.g. "update list.js" are not super descriptive ;-)

**Comment by pracplayopen** [comment]
sure, it's pretty simple not much more than first comment: "testing rest
serving w/website downloading (for redirect support)"

to elaborate, your original script leverages RSS to help make it work so
nicely.

this has advantage of making it simple but leaves out aws s3 file hosting
features accessible only via static website endpoint- such as http 301
redirects. if you're using a redirect to have "virtual files" that download
from a specific file or a file on another location on the internet, that
won't work w/your original script.

the suggested change works by seperating the url used for the listing from
that used for the download. if one commits this change and puts your static
s3 website endpoint into

BUCKET_WEBSITE_URL variable

in the same way that your existing variables are set, then all download
requests will go to the same file but with this base url. thus supporting
redirects and allowing virtual files.

**Comment by pracplayopen** [comment]
we use this to have a single virtual redirect.  we post this link in all
our external websites, and then update in one place (s3) when we have a new
file in that channel :

http://downloads.pracplay.com.s3-website-us-west-2.amazonaws.com/Glean/Featured/InstallLatestGlean

you can combine this with the cache-control feature in s3 to force caches
to timeout and repull.

**Comment by rufuspollock** [comment]
@pracplayopen ok - do any of your pull requests update the README with documentation and explanation of this new feature?

**Comment by pracplayopen** [comment]
thanks again for script.  you're welcome for the patch. i hope others find
it useful.
